presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -701,36 +703,47 @@ private TableFinishNode createTemporaryTableWrite(
             SchemaTableName temporaryTableName = metadata.getTableMetadata(session, tableHandle).getTable();
             InsertHandle insertHandle = new InsertHandle(insertTableHandle, new SchemaTableName(temporaryTableName.getSchemaName(), temporaryTableName.getTableName()));
 
+            PartitioningScheme partitioningScheme = new PartitioningScheme(
```

Comment:
Now I feel this "hard-coded" physical plan can sometimes tricky to maintain. Ideally we should only generate a simple logical plan (TableWriter -> TableFinish, TableScan). And use `AddExchange` and other rules to add the required `LocalExchange` / `TableWriteMerge` nodes. 

It looks like a major refactor though (needs to allow planner to allow multiple "logical plan fragments"), but might be worthwhile starting discussion / brainstorming -- we can do this refactor together with the CTE materialization project? cc @highker , @hellium01 , @rongrong